Feature/003 JDX wrapper wrapping around readJDX

sangttruong opened the pull request.

sangttruong committed:
Write wrapper function from readJDX package

- Convert the output from readJDX() functions from readJDX package to the corresponding hyperSpec object.

- Update documentation for additional import packages (add dplyr and readJDX), parameters, and return objects.

sangttruong committed:
Update DESCRIPTION file, include additional imports packages

sangttruong committed:
Update DESCRIPTION to resolve conflict with develop branch

bryanhanson reviewed (changes requested):
1. You should be the author of this function.  No need to mention me or CB.
2. Look at the other file import packages and name this function in a similar manner.  Perhaps `read_jdx`?
3. `# Case 1: Only includes spectrum` will be clearer if you make it `# Case 1: A single spectrum (IR, Raman, UV, processed/real 1D NMR, etc)`  Something along these lines should be added to the documentation too.
4. Case 3 can't occur and you can remove it.
5. Improve documentation generally.  For arguments basically the same as in `readJDX` feel free to direct users to `readJDX::readJDX` for further details (that way when I change `readJDX` you don't have to change much here).
6. For consideration... we might want to wait and ask the others.  But I think we should return the metadata (2nd list element from `readJDX`) for the user.  That would mean you always return a list, the first element of which could be the metadata and the remaining 2+ list elements the stuff you already return.  I'll put this on the agenda for Tuesday's meeting.

Commit:
Update gitignore to add DS_Store

Commit:
rename function to be consistent with other file import packages

Commit:
Update description for case 1 (single spectrum)

Commit:
Eliminate case 3 since hyperSpec cannot support 2D NMR

sangttruong committed:
Update read_jdx.R

- Update documentation of parameters (i,e, file, SOFC, and debug) and return value in read_jdx.R file.

- Update return value. Now return a list of metadata and hyperSpec object.

bryanhanson reviewed (changes requested):
1. For the documentation links to `readJDX`, use the method described [here](https://roxygen2.r-lib.org/articles/markdown.html#links)
2. I would keep one short sentence/phrase for each arg from `readJDX`: "The file to import", "Stop on Failed Check" and "The level of debug reporting"
3. Does this `return(list_jdx[[2]], spc)` work?  It must, but I've never seen it.  Better to make it a named list: `return(list(metadata = list_jdx[[2]], hyperSpec = spc))` These are much easier to use.

Commit:
Update documentation to include short description of arguments and format hyperlinks

Commit:
Reformat return to be easier to use
